databroker/broker.py

```diff
@@ -433,7 +433,12 @@ def add_filter(self, **kwargs):
         :meth:`Broker.clear_filters`
 
         """
-        self.filters.append(dict(**kwargs))
+        d = dict(**kwargs)
+        for val in self.filters:
+            for k,v in val.items():
+                if k in d.keys():
```

Comment:
`k in d` is equivalent to `k in d.keys()` and faster in Python 2 because it avoids creating an intermediate list.